File: frontend/src/model/Issue.ts
export type IssuePriority = "UNKNOWN" | "LOW" | "MEDIUM" | "HIGH" | "CRITICAL";
export type IssueStatus = "TODO" | "IN_PROGRESS" | "ON_HOLD" | "RESOLVED";
export type IssueType = "QUESTION" | "BUG" | "DOCUMENTATION" | "FEATURE";

import type { User } from "./User";

export interface Issue {
    id: number;
    title: string;
    description: string;
    type: IssueType;
    priority: IssuePriority;
    status: IssueStatus;
    imageUrl?: string | null;
    author: User;
    assignee?: User | null;
    resolvedAt: string | null;
    resolvedBy?: User | null;
    createdAt: string;
    updatedAt: string | null;
}

export interface CreateIssueRequest {
    title: string;
    description: string;
    type: IssueType;
    priority?: IssuePriority;
    assigneeEmail?: string;
    image?: File | null;
}

export interface ChangeIssueRequest {
    title?: string;
    description?: string;
    status?: IssueStatus;
    type?: IssueType;
    priority?: IssuePriority;
    assigneeEmail?: string;
}


export interface GetIssueRequest {
    type?: IssueType;
    status?: IssueStatus;
    priority?: IssuePriority;
    assigneeId?: number;
    authorId?: number;
    resolvedAfter?: string;
    resolvedBefore?: string;
    sortBy?: "createdAt" | "updatedAt" | "resolvedAt" | "priority" | "type" | "title";
    direction?: "asc" | "desc";
}

File: backend/src/main/java/com/bugboard26/backend/model/Issue.java
package com.bugboard26.backend.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;

@Entity
@Table(name = "issues")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Issue {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Column(nullable = false)
    private String title;

    @NotBlank
    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @NotNull
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private IssueType type;

    @NotNull
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private IssuePriority priority = IssuePriority.UNKNOWN;

    @NotNull
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private IssueStatus status = IssueStatus.TODO;

    @Column(nullable = true)
    private String imagePath;

    @NotNull
    @ManyToOne(optional = false)
    @JoinColumn(name = "author_id", nullable = false)
    @JsonIgnoreProperties({"passwordHash", "hibernateLazyInitializer", "handler"})
    private User author;

    @ManyToOne(optional = true)
    @JoinColumn(name = "assignee_id", nullable = true)
    @JsonIgnoreProperties({"passwordHash", "hibernateLazyInitializer", "handler"})
    private User assignee;

    @Column(nullable = true)
    private Instant resolvedAt;

    @NotNull
    @Column(nullable = false, updatable = false)
    private Instant createdAt;

    @Column(nullable = true)
    private Instant updatedAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = Instant.now();
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = Instant.now();
    }
}